spec.html

```diff
@@ -356,9 +360,13 @@
     1. If _initializer_ is not ~empty~, then
       1. Let _initValue_ be ? Call(_initializer_, _receiver_).
     1. Else, let _initValue_ be *undefined*.
-    1. If _name_ is a Private Name,
-      1. Perform ? PrivateFieldAdd(_name_, _O_, _initialValue_).
-    1. Otherwise, _name_ is a property key,
+    1. If _fieldName_ is a Private Name,
+      1. Perform ? PrivateFieldAdd(_fieldName_, _O_, _initialValue_).
+    1. Else,
+      1. NOTE: _fieldName_ is a property key.
+      1. If _fieldRecord_.[[IsAnonymousFunctionDefinition]] is *true*, then
+        1. Let _hasNameProperty_ be ? HasOwnProperty(_initValue_, `"name"`).
+        1. If _hasNameProperty_ is *false*, perform SetFunctionName(_initValue_, _propKey_).
```

Comment:
Oops, I just merged this patch but now I see that _propKey_ should be _fieldName_. I'll fix this when I add FNI for private names.